File: extension/build-zip.js
const fs = require('fs');
const path = require('path');
const pZlib = require('zlib');
const { promisify } = require('util');

// Super slim ZIP writer
async function createZip(sourceDir, destZip) {
    const out = fs.createWriteStream(destZip);

    // Write local file header
    const files = [];

    function walk(dir, prefix = '') {
        const entries = fs.readdirSync(dir);
        for (const entry of entries) {
            if (entry === '.DS_Store' || entry.startsWith('.')) continue;
            const fullPath = path.join(dir, entry);
            const relativePath = prefix ? prefix + '/' + entry : entry;

            if (fs.statSync(fullPath).isDirectory()) {
                walk(fullPath, relativePath);
            } else {
                files.push({ fullPath, relativePath });
            }
        }
    }

    walk(sourceDir);
    console.log(`Found ${files.length} files to ZIP`);

    // We will just use PowerShell again, but forcing Forward Slashes using 7zip if available, 
    // or just creating a Python script since Python's zipfile module is guaranteed to be POSIX compliant and built-in.
}

createZip();

File: extension/build-zip.py
import os
import zipfile

source_dir = r'e:\LAAG\extension\url-tracker'
dest_zip = r'e:\LAAG\extension\url-tracker-firefox.zip'

if os.path.exists(dest_zip):
    os.remove(dest_zip)

with zipfile.ZipFile(dest_zip, 'w', zipfile.ZIP_DEFLATED) as zipf:
    for root, dirs, files in os.walk(source_dir):
        # Ignore git/DS_Store
        if '.git' in root:
            continue
            
        for file in files:
            if file == '.DS_Store':
                continue
                
            file_path = os.path.join(root, file)
            # Create a relative path and force forward slashes for Firefox validation
            arcname = os.path.relpath(file_path, source_dir).replace('\\', '/')
            zipf.write(file_path, arcname)

print("Created POSIX-compliant ZIP archive:", dest_zip)
